util/views.py

```diff
@@ -0,0 +1,69 @@
+from django_filters.rest_framework import DjangoFilterBackend
+
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAdminUser
+from rest_framework.generics import ListAPIView
+from rest_framework.pagination import LimitOffsetPagination
+
+from rest_framework.renderers import (
+    BrowsableAPIRenderer,
+    JSONRenderer,
+)
+
+from .serializers import (
+    AllocationSerializer,
+    ProjectSerializer,
+    NestedProjectSerializer,
+)
+from .querysets import CCQuerySets
+from .filters import AllocationFilter, ProjectFilter
+
+
+class PaginationBaseView(LimitOffsetPagination):
+    default_limit = 100
+
+
+class ListBaseView(ListAPIView):
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [IsAdminUser]
+
+    pagination_class = PaginationBaseView
+    renderer_classes = [
+        BrowsableAPIRenderer,
+        JSONRenderer,
+    ]
+    filter_backends = [DjangoFilterBackend]
```

Comment:
This is probably the easiest place to start.
Any view inheriting from this will get:

1. session authentication + admin only permission
2. pagination with the defaults we've set above, or in the main config file
3. a browsable API view, that will expose whatever features the class is configured with
4. raw json output